src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -191,4 +201,82 @@ public void testReadsValidationStringencySilent(final String bamFileName, final
         testValidationStringency(bamFileName, referenceFileName, "SILENT", count);
     }
 
+    @Test(expectedExceptions = UserException.IncompatibleSequenceDictionaries.class)
+    public void testDisallowLexicographicallySortedVariantHeader_ifClashWithReference() throws Exception {
+        final GATKTool tool = new TestGATKToolWithFeatures();
+        final CommandLineParser clp = new CommandLineParser(tool);
+        final File vcfFile = new File(publicTestDir + "org/broadinstitute/hellbender/engine/lexicographically_sorted_dict.vcf");
+        final String[] args = {"--mask", vcfFile.getCanonicalPath(),
+                               "--reference", hg19MiniReference};
+        clp.parseArguments(System.out, args);
+
+        // This method throws despite the lexicographically-sorted sequence dictionary in the vcf.
+        //This is because the reference sequence dictionary clashes with the one from the VCF.
+        tool.onStartup();
+    }
+
+    @Test
+    public void testBestSequenceDictionary_fromReads() throws Exception {
+        final GATKTool tool = new TestGATKToolWithReads();
+        final CommandLineParser clp = new CommandLineParser(tool);
+        final File bamFile = new File(publicTestDir + "org/broadinstitute/hellbender/engine/reads_data_source_test1.bam");
+        final String[] args = {"-I", bamFile.getCanonicalPath()};
+        clp.parseArguments(System.out, args);
+        tool.onStartup();
+        //read the dict back in and compare to bam dict
+        final SAMSequenceDictionary toolDict = tool.getBestAvailableSequenceDictionary();
+        final SamReader open = SamReaderFactory.makeDefault().open(bamFile);
+        final SAMSequenceDictionary bamDict = open.getFileHeader().getSequenceDictionary();
+        toolDict.assertSameDictionary(bamDict);
+        bamDict.assertSameDictionary(toolDict);
+        Assert.assertEquals(toolDict, bamDict);
```

Comment:
Geez, do you really need all 3 asserts here to check that the dictionaries match? :)